### DISCUSSION TITLE
Can Link Type or Trees to have a cycle ?

### DISCUSSION THREAD

**Discussion by Jasminou** [airbnb/visx#1419] [Q&A] [answered]
Hi,

I wanted to know when using a Link Type or Trees

is it possible to have a cycle within the Tree/graphe ?

example: 
```
{
          name: "C",
          children: [
            {
              name: "C1"
            },
            {
              name: "D",
              children: [
                {
                  name: "D1"
                }
              ]
            }
          ]
 }
```

can C1 and D node be linked to the same D1 Node ?
![image](https://user-images.githubusercontent.com/14798195/150054819-d7e22f9b-4501-47bc-97cb-119d8762b6b7.png)

**Comment by williaster** [comment, marked as answer]
Hey @Jasminou 👋  thanks for checking out `visx`. The `@visx/hierarchy` package largely depends on [`d3-hierarchy`](https://github.com/d3/d3-hierarchy), and all of the hierarchical layouts available in d3 require a hierarchy structure where **every node has a single parent**. 

I think what you're describing is more of a true `network`, not a tree/hierarchy. So `@visx/network` would likely be what you need ... however we haven't ported over any of the d3 network layouts (which are harder to get right in `react`) so it may not be useful for that case.

**Reply by Jasminou** [reply]
Thank you @williaster
so with the network package I will need to manage the layout right and the position of the node

**Reply by williaster** [reply]
yep that's correct, so it's much more work than the Trees/Links unfortunately. You could use d3 packages for the math for the layout, and then still render in react.
